### PR TITLE
Add Template For Application Pages

### PR DESCRIPTION
**Pull request by oldwomanjosiah**
In this pull I add two templates and some helper classes used in rendering application pages (any where the user is already logged in)

## Templates Added:

- application_base.html
- partials/sidebar.html

### Modules
- TAScheduler.viewsupport  
  Will contain all helper classes needed to bridge the gap between application classes/code to template contexts
- TAScheduler.viewsupport.navbar  
  Contains helper classes for rendering the navbar, programatically define navbar items
- TAScheduler.viewsupport.icons  
  Contains helper classes for selecting glyphicons from bootstrap by name

### Classes
- navbar.NavbarItem
  - Contains all the relevant information needed for a single navbar row to be rendered (name, url, icon, enabled)
- navbar.AdminItems
  - Enum value which can be used as an iterable of NavbarItems which the Admin would see while they were logged in
  - Contains helper functions for highlighting & disabling whatever page the user is currently on so they cannot click it.
- icons.IconItem
  - Small class for turning the name of a bootstrap glyphicon into the classes needed to render it

Tests were also added for all of the above classes, all of which pass

### CODE REVIEW

**Comment by oldwomanjosiah** [comment]
[Create Application View Template](https://trello.com/c/0p3poHDN/20-create-application-view-template)

**Comment by oldwomanjosiah** [comment]
[Create Nav Bar Template](https://trello.com/c/wxH8JWnB/21-create-nav-bar-template)

**Review by tzortzos — APPROVED** [review]
everything looks fine to me from what i've seen, but i'd have others look over it as well before merging since these are big changes

File: project/TAScheduler/viewsupport/test_icons.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    def test_returns_name(self):
+        icon = IconItem('home')
+        self.assertEqual(icon.take(), 'home', msg='Did not return correct member')
+
+    def test_returns_classes(self):
+        icon = IconItem('home')
+        self.assertEqual(icon.classes(), 'glyphicon glyphicon-home')
+
+
+if __name__ == '__main__':
```
Comment (lnahnan) [inline project/TAScheduler/viewsupport/test_icons.py:19]: These can be deleted with no "main" actually being run

**Review by lnahnan — COMMENTED** [review]
Replace example html with app html
delete calls to main

**Review by lnahnan — APPROVED** [review]

**Pull request merged**